src/converter/jfr2pprof.java

```diff
@@ -209,19 +244,49 @@ private byte[] getMethodName(final long methodId, final byte methodType) {
     }
 
     public static void main(String[] args) throws Exception {
-        if (args.length < 2) {
-            System.out.println("Usage: java " + jfr2pprof.class.getName() + " input.jfr output.pprof");
+
+        final Arguments arguments = new Arguments(args);
+        if (arguments.input == null || arguments.output == null) {
+            System.out.println("Usage: java " + jfr2pprof.class.getName() + " [options] input.jfr output.pprof");
+            System.out.println();
+            System.out.println("options can be one or more of the following:");
+            System.out.println("  --cpu        CPU");
+            System.out.println("  --alloc      Allocation");
+            System.out.println("  --lock       Lock contention");
+            System.out.println();
+            System.out.println("If no options are given, --cpu is assumed");
             System.exit(1);
         }
 
-        File dst = new File(args[1]);
-        if (dst.isDirectory()) {
-            dst = new File(dst, new File(args[0]).getName().replace(".jfr", ".pprof"));
+        final List<String> types = new ArrayList<>();
+
+        if (arguments.cpu) {
+            types.add(TYPE_CPU);
+        }
+
+        if (arguments.alloc) {
+            types.add(TYPE_ALLOC);
+        }
+
+        if (arguments.lock) {
+            types.add(TYPE_LOCK);
         }
 
-        try (final JfrReader jfr = new JfrReader(args[0]);
-             final FileOutputStream out = new FileOutputStream(dst)) {
-            new jfr2pprof(jfr).dump(out);
+        if (types.isEmpty()) {
+            types.add(TYPE_CPU);
+        }
+
+        for (final String type : types) {
+          File dst = new File(arguments.output + "." + type);
+
+          if (dst.isDirectory()) {
+              dst = new File(dst, new File(arguments.input).getName().replace(".jfr", ".pprof" + "." + type));
```

Comment:
I'd put `type` ahead of the file extension and only for the case when multiple types are given.